### PR TITLE
DATAUP-678: New and moved methods in the JobManager module

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Minor reorganisation of the Job Manager module, and addition of two new batch job methods -- request info and request status.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-678
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 1]
```diff
@@ -748,60 +748,9 @@ define([
                     validStatuses: Jobs.validStatusesForAction.retry,
                 });
             }
-
-            /**
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:752]: move these down into the batch mixin

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 2]
```diff
@@ -892,15 +841,137 @@ define([
...
+                }
+            }
+
+            /**
+             * Cancel a batch job by submitting a cancel request for the batch job container
+             *
+             * This action is triggered by hitting the 'Cancel' button at the top right of the
+             * bulk cell
+             */
+            cancelBatchJob() {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:920]: moved from above -- no code changes

File: kbase-extension/static/kbase/js/common/jobManager.js [hunk 3]
```diff
@@ -748,60 +748,9 @@ define([
...
-                    this.bus.emit('reset-cell', {
-                        cellId: this.cellId,
-                        ts: Date.now(),
-                    });
-                }
-            }
         };
 
-    const BatchInitMixin = (Base) =>
+    const BatchMixin = (Base) =>
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobManager.js:753]: renamed to be more general

File: test/unit/spec/common/jobManager-Spec.js [hunk 4]
```diff
@@ -2152,5 +2079,181 @@ define([
...
+
+            it('does not make a request if the batch job is terminal', function () {
+                // update the batch job to terminated
+                this.jobManagerInstance.model.setItem('exec.jobState', {
+                    ...this.jobManagerInstance.model.getItem('exec.jobState'),
+                    status: 'terminated',
+                });
+                spyOn(this.jobManagerInstance.bus, 'emit');
+                this.jobManagerInstance.requestBatchStatus();
+                expect(this.bus.emit.calls.allArgs()).toEqual([]);
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:2121]: no job status request if the batch job has been terminated

File: test/unit/spec/common/jobManager-Spec.js [hunk 5]
```diff
@@ -2152,5 +2079,181 @@ define([
...
+
+            it('does not make a request if there are no active jobs', function () {
+                const terminalJobs = JobsData.allJobs.filter((job) => {
+                    return !ACTIVE_JOB_IDS.includes(job.job_id);
+                });
+                terminalJobs.push(JobsData.batchParentJob);
+                Jobs.populateModelFromJobArray(this.jobManagerInstance.model, terminalJobs);
+                spyOn(this.jobManagerInstance.bus, 'emit');
+                this.jobManagerInstance.requestBatchStatus();
+                expect(this.bus.emit.calls.allArgs()).toEqual([]);
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:2110]: no job status request if the batch parent job is the only non-terminal job in the batch

File: test/unit/spec/common/jobManager-Spec.js [hunk 6]
```diff
@@ -2152,5 +2079,181 @@ define([
...
+            beforeEach(function () {
+                this.jobManagerInstance = createJobManagerInstance(
+                    this,
+                    BatchMixin(JobManagerCore)
+                );
+            });
+
+            it('requests info for the batch if none is present', function () {
+                spyOn(this.jobManagerInstance.bus, 'emit');
+                this.jobManagerInstance.requestBatchInfo();
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:2135]: no `exec.jobs.info` at all in the model

File: test/unit/spec/common/jobManager-Spec.js [hunk 7]
```diff
@@ -2152,5 +2079,181 @@ define([
...
+
+                this.jobManagerInstance.model.setItem('exec.jobs.info', {});
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:2137]: `exec.jobs.info` exists, but it is empty

File: test/unit/spec/common/jobManager-Spec.js [hunk 8]
```diff
@@ -2152,5 +2079,181 @@ define([
...
+                ]);
+            });
+        });
+
+        const resetJobsCallArgs = [
+            jcm.MESSAGE_TYPE.STOP_UPDATE,
+            { [jcm.PARAM.BATCH_ID]: BATCH_ID },
+        ];
+
+        describe('cancelBatchJob', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobManager-Spec.js:2180]: the moved tests

**Review by briehl — APPROVED** [review]
🎉 
It's so clean and straight forward!

**Pull request merged**